nest/src/RoycoNestMarketHelper.sol

```diff
@@ -0,0 +1,482 @@
+            $.vaultExists[identifier] = true;
+        }
+
+        // Update the vault
+        $.vaults[identifier] = BoringVault({
+            teller: ITeller(teller),
+            vault: IBoringVault(vault),
+            accountant: IAccountantWithRateProviders(accountant),
+            slippageBps: slippageBps,
+            performanceBps: performanceBps,
+            active: true
+        });
+
+        emit VaultUpdated(identifier);
+    }
+
+    /**
+     * @dev Update fees for a vault
+     */
+    function updateVaultFees(
+        string memory identifier,
+        uint256 slippageBps,
+        uint256 performanceBps
+    ) external onlyRole(ADMIN_ROLE) {
+        RoycoMarketHelperStorage storage $ = _getRoycoMarketHelperStorage();
+
+        if (!$.vaults[identifier].active) {
+            revert VaultNotActive(identifier);
+        }
+
+        $.vaults[identifier].slippageBps = slippageBps;
+        $.vaults[identifier].performanceBps = performanceBps;
+
+        emit VaultUpdated(identifier);
+    }
+
+    /**
+     * @dev Get vault details
+     */
+    function vaults(
+        string memory identifier
+    )
+        external
+        view
+        returns (
+            address teller,
+            address vault,
+            address accountant,
+            uint256 slippageBps,
+            uint256 performanceBps,
+            bool active
+        )
+    {
+        RoycoMarketHelperStorage storage $ = _getRoycoMarketHelperStorage();
+        BoringVault storage boringVault = $.vaults[identifier];
+
+        return (
+            address(boringVault.teller),
+            address(boringVault.vault),
+            address(boringVault.accountant),
+            boringVault.slippageBps,
+            boringVault.performanceBps,
+            boringVault.active
+        );
+    }
+
+    /**
+     * @dev Calculate minimum mint amount based on vault parameters
+     */
+    function calculateMinimumMint(
+        string memory identifier,
+        address depositAsset,
+        uint256 depositAmount
+    ) public view returns (uint256) {
+        RoycoMarketHelperStorage storage $ = _getRoycoMarketHelperStorage();
+
+        BoringVault storage boringVault = $.vaults[identifier];
+
+        if (!boringVault.active) {
+            revert VaultNotActive(identifier);
+        }
+
+        // Get rate from Accountant using the depositAsset
+        // Note: rate already includes decimals (e.g., 2e18 for 2 USD)
+        uint256 rate = boringVault.accountant.getRateInQuote(IERC20(depositAsset));
+        if (rate == 0) {
+            revert InvalidRate();
+        }
+
+        // Get vault decimals
+        uint8 vaultDecimals = boringVault.vault.decimals();
+
+        // Calculate raw mint amount with precision by multiplying first
+        // Solidity 0.8+ will revert on overflow automatically
+        uint256 decimalsMultiplier = 10 ** vaultDecimals;
+        uint256 rawMintAmount = (depositAmount * decimalsMultiplier) / rate;
```

Comment:
Suggestion: divide by rate at the very end (`return rawMintAmount / rate`, `return minimumMint / rate`) to reduce rounding errors